clientlibs/java/src/main/java/org/upgradeplatform/client/ExperimentClient.java

```diff
@@ -55,8 +56,19 @@ public ExperimentClient(String userId, String authToken, String baseUrl, Map<Str
 			throw new IllegalArgumentException(INVALID_STUDENT_ID);
 		}
 		this.userId = userId;
+		this.apiService = new APIService(baseUrl, authToken, UUID.randomUUID().toString(), properties);
```

Comment:
you don't need to copy-paste the whole body of the constructor.  Just do:

```
    this(userId, authToken, UUID.randomUUID().toString(), baseUrl, properties);
```

Reply:
good catch, it's been updated